OpenRA.Mods.Common/Widgets/Logic/MultiplayerLogic.cs

```diff
@@ -322,7 +329,13 @@ void RefreshServerList()
 					}
 				}
 
-				Game.RunAfterTick(() => RefreshServerListInner(games));
+				var lanGames = lanGameLocations.Select(bl => {
+						var gs = new GameServer(MiniYaml.FromString(bl.Data)[0].Value);
+						gs.Address = bl.Address.ToString().Split(':')[0] + ":" + gs.Address.Split(':')[1];
```

Comment:
Making the address non-readonly and then manipulating it here isn't very nice.  Please modify the yaml given to the GameServer instead:
```csharp
bl =>
{
	var game = MiniYaml.FromString(bl.Data)[0].Value;

	// Rewrite the server address with the correct IP
	var addressNode = game.Nodes.FirstOrDefault(n => n.Key == "Address");
	if (addressNode != null)
		addressNode.Value.Value = bl.Address.ToString().Split(':')[0] + ":" + addressNode.Value.Value.Split(':')[1];

	return new GameServer(game);
}
```

Also note a style nit: opening braces for a block should be on their own line.